compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -626,11 +627,16 @@ public void visit(BLangAnnotationAttachment astAnnotAttach) {
     }
 
     private boolean isCompileTimeAnnotationValue(BLangExpression expr) {
-        // TODO Compile time literal constants
         switch (expr.getKind()) {
             case LITERAL:
             case NUMERIC_LITERAL:
                 return true;
+            case INVOCATION:
+                BLangInvocation invocation = (BLangInvocation) expr;
+                if (invocation.name.getValue().equals(CLONE_READ_ONLY)) {
```

Comment:
Thinking about this a little more, currently an invocation is not a compile-time-known value right?

So I think we should differentiate between the `.cloneReadOnly` we add, and what the user may have added for a field.

Shall we allow this only when this is the attachment itself, and return false for recursive checks?